core/controllers/platform_feature_test.py

```diff
@@ -249,97 +239,16 @@ def _mock_dummy_feature_stage(
             param_list.ParamNames.DUMMY_FEATURE_FLAG_FOR_E2E_TESTS.value)
         return self.swap(feature, '_feature_stage', stage.value)
 
-    def test_get_with_dummy_feature_enabled_in_dev_returns_ok(self) -> None:
```

Comment:
There still need to be tests for dev and prod mode -- i.e. the server mode. The difference now is that dev/prod mode behaviour is determined based on the current "feature stage" of the flag and the current server mode, rather than an explicit "server mode" filter.

Reply:
Yes, that is correct. Please note that this feature is the e2e_dummy_feature which is in prod mode now and we just want to test the functionality here. As the feature is in prod mode this is available in all the server stage.
The tests related to the server modes and feature stages are present in the `feature_services_test` file. We can say that we were simply repeating the tests with e2e_dummy_feature in this file so I have removed this part.
Let me know what do you think.

Reply:
I'm OK with anything as long as (a) we have full coverage of behaviours, (b) the test name correctly describes what the test is testing, (c) the test is the simplest test possible to test whatever it's testing.